src/process.cr

```diff
@@ -507,6 +507,150 @@ class Process
     io.close if io
   end
 
+  # Returns the **real** user id of the current process.
+  def self.user_id : LibC::UidT
+    LibC.getuid
+  end
+
+  # Returns the **effective** user id of the current process.
+  def self.effective_user_id : LibC::UidT
+    LibC.geteuid
+  end
+
+  # Returns the **real** group id of the current process.
+  def self.group_id : LibC::GidT
+    LibC.getgid
+  end
+
+  # Returns the **effective** group id of the current process.
+  def self.effective_group_id : LibC::GidT
+    LibC.getegid
+  end
+
+  # :nodoc:
+  UID_DEFAULT = if LibC::UidT.new(0).is_a?(Int::Signed)
+                  LibC::UidT.new(-1)
+                else
+                  LibC::UidT::MAX
+                end
+  GID_DEFAULT = if LibC::GidT.new(0).is_a?(Int::Signed)
+                  LibC::GidT.new(-1)
+                else
+                  LibC::GidT::MAX
+                end
+
+  # Permanently transition to another account.
+  #
+  # user_id's, group_id's and groups are changed to the account provided.
+  #
+  # Call `chroot` or other privileged operations becore calling this method.
+  #
+  # Example:
+  #
+  # ```
+  # user = System::User.find_by name: "crystal"
+  # Process.become user
+  #
+  # user = System::User.find_by name: "crystal"
+  # group = System::Group.find_by name: "wheel"
+  # Process.become user, group # Use a different group other than the user's.
+  # ```
+  def self.become(user : System::User, group : System::Group? = nil)
+    group ||= user.group
+    # TODO setgroups
```

Comment:
Could you elaborate on this TODO? I don't understand what it means.

Reply:
It means [setgroups](https://linux.die.net/man/2/setgroups) should be called before `become_user`, but the `User` class doesn't provide a group list.

Also, the API and naming was never discussed.  I would use `Process.groups=` before `become_` was settled on.   Now I don't know what to call it.

Reply:
All right. This extensive explanation should be added to the comment.